### DISCUSSION TITLE
Need an option in .NET MAUI Blazor Hybrid to do something after the Razor component is rendered

### DISCUSSION THREAD

**Discussion by egvijayanand** [dotnet/maui#18936] [General]
Trying to link Razor component and .NET MAUI platform (Hybrid).

Currently, only BlazorWebView **Initializing** and **Initialized** events are available, which work on the underlying platform **WebView**.

The newly introduced **TryDispatchAsync()** method establishes the link from MAUI to Razor, invocation to happen on the MAUI end. And I'm looking in the opposite direction. Once the Razor component of the **StartPath** property (at least) needs to interact with the MAUI code.

Is there any option available to achieve this? @Eilon

**Comment by Eilon** [comment]
Hi @egvijayanand , can you give an example of what you'd like the app to do after the Razor component is rendered? There could be different solutions depending on what the scenario is.

**Reply by egvijayanand** [reply]
To keep it simple, would like to access the scoped **ServiceProvider** instance to get access to the shared state. The same way TryDispatchAsync() works. The problem now is the interaction has to get initiated from MAUI. And what I try is, it can start from either side so they work in tandem.

Tried **Initialized**, but the Component is not rendered at this time.

Maybe, it's time to define another event or any workaround available?

**Reply by egvijayanand** [reply]
Hi @Eilon,

Thanks for your response.

Have found a way to make it work using the [Messaging](https://learn.microsoft.com/en-us/dotnet/architecture/maui/communicating-between-components) technique.

After the Razor component is rendered, will send out a message with the payload (can be any object), as the message handler is registered on the MAUI side. It receives that and it works as expected.

**Reply by Eilon** [reply]
I posted a sample at https://github.com/Eilon/Maui18936AfterBlazorRenderApp, and in particular this small commit: https://github.com/Eilon/Maui18936AfterBlazorRenderApp/commit/5b57aabc148a25304c83cc9556803e7b195eb26e

In this sample I create a service that has an event that different parts of the app can listen to. And in the Razor layout page I raise the event after rendering.

This is a very simple pattern that can be adapted to any app's needs.

**Reply by egvijayanand** [reply]
Let me check and get back.
